src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslCachedSystemStoreProvider.cs

```diff
@@ -283,6 +282,35 @@ bool ProcessFile(string file, out DateTime lastModified, bool skipStat = false)
                 }
 
                 return readData;
+
+                static bool TryReadPemOrDer(SafeBioHandle fileBio, [NotNullWhen(true)] out ICertificatePal? pal)
+                {
+                    int currentPosition = Interop.Crypto.BioTell(fileBio);
+
+                    if (OpenSslX509CertificateReader.TryReadX509PemNoAux(fileBio, out pal))
+                    {
+                        return true;
+                    }
+
+                    // If BIO_tell was not able to tell us the position, then the BIO cannot be rewound. Treat this the
+                    // same as not being able to read the contents as DER.
+                    if (currentPosition < 0)
+                    {
+                        pal = null;
+                        return false;
+                    }
+
+                    int seek = Interop.Crypto.BioSeek(fileBio, currentPosition);
+
+                    // If the BIO_seek failed, treat this the same as not being able to read the contents as DER.
+                    if (seek < 0)
+                    {
+                        pal = null;
+                        return false;
+                    }
+
+                    return OpenSslX509CertificateReader.TryReadX509Der(fileBio, out pal);
```

Comment:
I think we definitely don't want to read a DER following a PEM, because then we're way off in fairy-land (this function is OK by itself, but it's now in a loop, so it'll allow multi-DER and/or PEM/DER/PEM/DER/...).

Which, yeah, the old code sort of implied, but, it didn't work :smile:.

I think the better answer is to just delete the or-DER.  No one has complained about it not working, which means my original "maybe it's DER on some distro?" hasn't manifested.